homeassistant/components/unifi/update.py

```diff
@@ -23,11 +34,90 @@
 from .const import ATTR_MANUFACTURER, DOMAIN as UNIFI_DOMAIN
 
 if TYPE_CHECKING:
+    from aiounifi.models.event import EventKey
+
     from .controller import UniFiController
 
+_DataT = TypeVar("_DataT", bound=APIItem)
+_HandlerT = TypeVar("_HandlerT", bound=APIHandler)
+
+Subscription = Callable[[CallbackType, ItemEvent], UnsubscribeType]
+
 LOGGER = logging.getLogger(__name__)
 
-DEVICE_UPDATE = "device_update"
+
+@callback
+def async_device_available_fn(controller: UniFiController, obj_id: str) -> bool:
+    """Check if device is available."""
+    device = controller.api.devices[obj_id]
+    return controller.available and not device.disabled
+
+
+async def async_device_control_fn(api: aiounifi.Controller, obj_id: str) -> None:
+    """Control upgrade of device."""
+    await api.request(DeviceUpgradeRequest.create(obj_id))
+
+
+@callback
+def async_device_device_info_fn(api: aiounifi.Controller, obj_id: str) -> DeviceInfo:
+    """Create device registry entry for device."""
+    device = api.devices[obj_id]
+    return DeviceInfo(
+        connections={(CONNECTION_NETWORK_MAC, device.mac)},
```

Comment:
This was previously set to `obj_id`. Are `obj_id` and `mac` identical? or does it need a migration step?
```python
        connections={(CONNECTION_NETWORK_MAC, obj_id)},
```

Reply:
It's just a normalization. The primary key "obj_id" is Mac for devices and clients.